pkg/controllers/scheduling/clustersetbinding_event_handler.go

```diff
@@ -0,0 +1,68 @@
+package scheduling
+
+import (
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+
+	clusterlisterv1alpha1 "github.com/open-cluster-management/api/client/cluster/listers/cluster/v1alpha1"
+	clusterapiv1alpha1 "github.com/open-cluster-management/api/cluster/v1alpha1"
+)
+
+type clusterSetBindingEventHandler struct {
+	clusterSetLister     clusterlisterv1alpha1.ManagedClusterSetLister
+	placementLister      clusterlisterv1alpha1.PlacementLister
+	enqueuePlacementFunc enqueuePlacementFunc
+}
+
+func (h *clusterSetBindingEventHandler) OnAdd(obj interface{}) {
+	h.onChange(obj)
+}
+
+func (h *clusterSetBindingEventHandler) OnUpdate(oldObj, newObj interface{}) {
+	h.onChange(newObj)
+}
+
+func (h *clusterSetBindingEventHandler) OnDelete(obj interface{}) {
+	h.onChange(obj)
+}
+
+func (h *clusterSetBindingEventHandler) onChange(obj interface{}) {
+	// skip resource other than ManagedClusterSetBinding
+	clusterSetBinding, ok := obj.(*clusterapiv1alpha1.ManagedClusterSetBinding)
+	if !ok {
+		return
+	}
+
+	_, err := h.clusterSetLister.Get(clusterSetBinding.Name)
+	// skip if the clusterset does not exist
+	if errors.IsNotFound(err) {
+		return
+	}
+	if err != nil {
+		klog.Errorf("Unable to get clusterset %q: %v", clusterSetBinding.Name, err)
+		return
+	}
+
+	err = enqueuePlacementsInNamespace(clusterSetBinding.Namespace, h.placementLister, h.enqueuePlacementFunc)
+	if err != nil {
+		klog.Errorf("Unable to enqueue placements in namespace %q: %v", clusterSetBinding.Namespace, err)
+	}
+}
+
+// enqueuePlacementsInNamespace enqueues all placements under a particular namespace into controller queue
+// for further reconciliation
+func enqueuePlacementsInNamespace(
+	namespace string,
+	placementLister clusterlisterv1alpha1.PlacementLister,
+	enqueuePlacementFunc enqueuePlacementFunc,
+) error {
+	placements, err := placementLister.Placements(namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	for _, placement := range placements {
+		enqueuePlacementFunc(placement.Namespace, placement.Name)
```

Comment:
should we also filter binding name here?

Reply:
Added logic to filter placement with binding name.